### PR TITLE
Bugfix: Improve Ready and Health Checks

### PR DESCRIPTION
**Pull request by fschade**
## Description

- Fixed the issue where ready and health checks were not functioning correctly.
- Enhanced the checks to verify the readiness and health of individual services by evaluating their specific requirements.

## Service Probes

Checklist moved to #10281 

## Questions

- how to handle services that do not have a http endpoint

## Related Issue
- Fixes https://github.com/owncloud/ocis/issues/8783

## Motivation and Context
health and ready checks are essential to find out if a service is ready to serve requests. In the past, the checks were not functioning correctly. This release fixes the issue and enhances the checks to verify the readiness and health of individual services by evaluating their specific requirements.

## How Has This Been Tested?
- unit tests
- local instance

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [X] Code changes
- [X] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by butonic** [comment]
related: https://github.com/owncloud/ocis/issues/6774

**Comment by dragonchaser** [comment]
We decided to split the issue into several PRs, this PR only contains 3 service healthchecks and basic skeletons & refactoring, the others will be implemented in follow-up PRs. (List see #10281 )

File: ocis-pkg/handlers/checktcp.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import (
+	"context"
+	"net"
+)
+
+// NewTCPCheck returns a check that connects to a given tcp endpoint.
+func NewTCPCheck(address string) func(ctx context.Context) error {
+	return func(_ context.Context) error {
+		conn, err := net.Dial("tcp", address)
```
Comment (2403905) [inline ocis-pkg/handlers/checktcp.go:11]: Does it make sense to set the timeout?

Comment (dragonchaser) [reply ocis-pkg/handlers/checktcp.go:11]: @2403905 good point.... will look into it

File: ocis-pkg/handlers/checker.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+		status = h.conf.statusFailed
+		h.conf.logger.Error().Err(err).Msg("check failed")
+	}
+
+	w.Header().Set("Content-Type", "text/plain") // fixMe: should be application/json!?!
+	w.WriteHeader(status)
+
+	if _, err := io.WriteString(w, http.StatusText(status)); err != nil { // io.WriteString should not fail, but if it does, we want to know.
+		h.conf.logger.Panic().Err(err).Msg("failed to write response")
+	}
```
Comment (butonic) [inline ocis-pkg/handlers/checker.go:116]: no content type needed as per https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request kubernetes is only interested in the status code, the rest is user info:
```go
http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
    duration := time.Now().Sub(started)
    if duration.Seconds() > 10 {
        w.WriteHeader(500)
        w.Write([]byte(fmt.Sprintf("error: %v", duration.Seconds())))
    } else {
        w.WriteHeader(200)
        w.Write([]byte("ok"))
    }
})
```
> [...]  To perform a probe, the kubelet sends an HTTP GET request to the server that is running in the container and listening on port 8080. If the handler for the server's /healthz path returns a success code, the kubelet considers the container to be alive and healthy. If the handler returns a failure code, the kubelet kills the container and restarts it.
>
> Any code greater than or equal to 200 and less than 400 indicates success. Any other code indicates failure.

File: services/antivirus/pkg/server/debug/server.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	readyHandler := handlers.NewCheckHandler(
+		handlers.NewCheckHandlerConfiguration().
+			WithLogger(options.Logger).
+			WithCheck("nats reachability", handlers.NewNatsCheck(options.Config.Events.Cluster)).
+			WithCheck("antivirus reachability", func(ctx context.Context) error {
+				cfg := options.Config
+				switch cfg.Scanner.Type {
+				default:
+					// there is not av configured, so we panic
+					panic("no antivirus configured")
```
Comment (butonic) [inline services/antivirus/pkg/server/debug/server.go:31]: hm ... why panic?

Comment (dragonchaser) [reply services/antivirus/pkg/server/debug/server.go:31]: Because antivirus was explicitly enabled and forgotten to be configured.

**Review by butonic — APPROVED** [review]

**Pull request merged**